### PR TITLE
suggestions for effect plugin

### PR DESCRIPTION
**Pull request by tim-smart**

### CODE REVIEW

File: plugin/effect/src/index.ts [hunk 1]
```diff
@@ -1,21 +1,20 @@
-import { Effect } from "effect";
+import * as Effect from "effect/Effect";
```
Comment (tim-smart) [inline plugin/effect/src/index.ts:1]: alias imports are better for library code, as they cause less issues with bundlers

File: plugin/effect/src/index.ts [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
 import * as z from "zod";
 
 function zodEffect(this: z.ZodType, data: unknown, params?: any) {
-  return Effect.tryPromise({
-    try: async () => this.parseAsync(data, params),
-    catch(error) {
-      return error as z.ZodError;
-    },
-  });
+  return Effect.flatMap(
```
Comment (tim-smart) [inline plugin/effect/src/index.ts:5]: Effect.promise + flatMap allows us to use the safe apis

File: plugin/effect/src/index.ts [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
+  );
 }
 
 function zodEffectSync(this: z.ZodType, data: unknown, params?: any) {
-  return Effect.try({
-    try: () => this.parse(data, params),
-    catch(error) {
-      return error as z.ZodError;
-    },
+  return Effect.suspend(() => {
```
Comment (tim-smart) [inline plugin/effect/src/index.ts:13]: Effect.suspend allows us to wrap a side effect that returns an Effect, perfect for the sync api.

**Comment by colinhacks** [comment]
Thanks!!

**Pull request merged**